src/components/voyagecalculator_iap.tsx

```diff
@@ -0,0 +1,388 @@
+							</b>
+						</Form.Field>
+					</Form.Group>
+
+					<Form.Group>
+						<Form.Field
+							control={Dropdown}
+							clearable
+							fluid
+							multiple
+							search
+							selection
+							options={this.state.peopleList[peopleListStyle]}
+							placeholder='Select or search for crew'
+							label={
+								"Crew you don't want to consider for voyage" +
+								(this.state.activeEvent ? ` (preselected crew which gives bonus in the event ${this.state.activeEvent})` : '')
+							}
+							value={this.state.currentSelection}
+							onChange={(e, { value }) => this.setState({ currentSelection: value })}
+						/>
+					</Form.Group>
+
+					<Form.Group inline>
+						<Form.Field
+							control={Checkbox}
+							label='Consider active (on shuttles) crew'
+							checked={this.state.includeActive}
+							onChange={(e, { checked }) => this.setState({ includeActive: checked })}
+						/>
+
+						<Form.Field
+							control={Checkbox}
+							label='Consider frozen (vaulted) crew'
+							checked={this.state.includeFrozen}
+							onChange={(e, { checked }) => this.setState({ includeFrozen: checked })}
+						/>
+					</Form.Group>
+
+					{this.state.result && (
+						<React.Fragment>
+							<p>
+								Estimated duration: <b>{formatTimeSeconds(this.state.result.score * 60 * 60)}</b>
+							</p>
+							<ul>
+								{this.state.result.entries.map((entry, idx) => {
+									let pcrew = crew.find(c => c.id === entry.choice);
+									let acrew = playerData.player.character.crew.find(c => c.symbol === pcrew.symbol);
+									return (
+										<li key={idx}>
+											{voyageData.voyage_descriptions[0].crew_slots[entry.slotId].name}
+											{'  :  '}
+											<CrewPopup crew={acrew} />
+										</li>
+									);
+								})}
+							</ul>
+						</React.Fragment>
+					)}
+
+					<Form.Group>
+						<Form.Button
+							primary
+							onClick={() => this._calcVoyageData(bestShip.score)}
+							disabled={this.state.calcState === CalculatorState.InProgress}>
+							Calculate best crew selection
+						</Form.Button>
+					</Form.Group>
+				</Form>
+			</div>
+		);
+	}
+
+	_bestVoyageShip(ships: any[], voyageData: any): any[] {
+		let voyage = voyageData.voyage_descriptions[0];
+
+		let consideredShips: any[] = [];
+		ships.forEach((ship: any) => {
+			let entry = {
+				ship: ship,
+				score: ship.antimatter
+			};
+
+			if (ship.traits.find((trait: any) => trait == voyage.ship_trait)) {
+				entry.score += 150; // TODO: where is this constant coming from (Config)?
+			}
+
+			consideredShips.push(entry);
+		});
+
+		consideredShips = consideredShips.sort((a, b) => b.score - a.score);
+
+		return consideredShips;
+	}
+
+	_bonusCrewForCurrentEvent(eventData: any[], crewlist: any[]): BonusCrew | undefined {
+		let result = new BonusCrew();
+
+		if (eventData && eventData.length > 0) {
+			let activeEvent = eventData
+			  .filter((ev) => (ev.seconds_to_end > 0))
```

Comment:
While we're making changes here, we should ignore events over a day away.

```suggestion
			  .filter((ev) => (ev.seconds_to_end > 0 && ev.seconds_to_start < 86400))
```